### PR TITLE
feat(cicd): add release pipeline

### PR DESCRIPTION
**Pull request by paologallinaharbur**
Fix https://github.com/newrelic/k8s-metadata-injection/issues/141

### CODE REVIEW

File: .github/ct.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Chart linter defaults to `master` branch so we need to specify this as the default branch
+# or `cl` will fail with a not-so-helpful error that says:
+# "Error linting charts: Error identifying charts to process: Error running process: exit status 128"
+target-branch: main
```
Comment (paologallinaharbur) [inline .github/ct.yaml:4]: NL I'll fix that with other nits

File: .github/workflows/release-integration.yml [hunk 2]
```diff
@@ -43,36 +43,43 @@ jobs:
...
       - name: Build and push docker image
+        if: ${{ github.event.release.prerelease }}
         run: |
+          DOCKER_IMAGE_TAG=${DOCKER_IMAGE_TAG}-pre"
           docker buildx build --push --platform=$DOCKER_PLATFORMS \
             -t $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG \
             .
-      - name: Push :latest image
-        if: ${{ ! github.event.release.prerelease }}
+      - name: Push release image
```
Comment (kang-makes) [inline .github/workflows/release-integration.yml:80]:
```suggestion
      - name: Push release image
        if: ${{ ! github.event.release.prerelease }}
```

File: .github/ct.yaml (hunk 1 above)
Comment (kang-makes) [reply .github/ct.yaml:4]: NIT

**Review by kang-makes — COMMENTED** [review]
`latest` is always released :/

File: .github/workflows/release-integration.yml (hunk 2 above)
Comment (paologallinaharbur) [reply .github/workflows/release-integration.yml:80]: you are totally right!

**Review by kang-makes — APPROVED** [review]
LGTM

**Pull request merged**